src/control/server/mgmt_pool.go

```diff
@@ -543,20 +543,24 @@ func (svc *mgmtSvc) PoolDestroy(ctx context.Context, req *mgmtpb.PoolDestroyReq)
 		ds := drpc.DaosStatus(evresp.Status)
 		svc.log.Debugf("MgmtSvc.PoolDestroy drpc.MethodPoolEvict, evresp:%+v\n", evresp)
 
+		// If the destroy request is being forced, we should additionally zap the label
+		// so the entry doesn't prevent a new pool with the same label from being created.
+		if req.Force {
+			ps.PoolLabel = ""
+		}
+
 		// Transition pool state (unless evict returned busy, and not force destroying).
-		if !(ds == drpc.DaosBusy && !req.Force) {
+		if req.Force || ds != drpc.DaosBusy {
 			ps.State = system.PoolServiceStateDestroying
+		}
+
+		// Ensure that the label and/or state updates are persisted as appropriate.
+		if req.Force || ds != drpc.DaosBusy {
```

Comment:
This could all be in the same block, looks like the condition is the same

Reply:
Hmm, yep. I changed the test because the inverted booleans were making my head hurt, but you're right these two can be collapsed into one.